sh/common_gas.sh

```diff
@@ -0,0 +1,52 @@
+# Common gas price and gas limit utilities
+
+# set minimum gas price (mostly for Arbitrum and BNB)
+declare -i min_gas_price
+min_gas_price="$(get_config minGasPriceGwei)"
+min_gas_price=$((min_gas_price * 1000000000))
+declare -r -i min_gas_price
+
+declare -i gas_price
+gas_price="$(cast gas-price --rpc-url "$rpc_url")"
+if (( gas_price < min_gas_price )) ; then
+    echo 'Setting gas price to minimum of '$((min_gas_price / 1000000000))' gwei' >&2
+    gas_price=$min_gas_price
+fi
+declare -r -i gas_price
+
+declare -i gas_estimate_multiplier
+gas_estimate_multiplier="$(get_config gasMultiplierPercent)"
+declare -r -i gas_estimate_multiplier
+
+# EIP-7825 gas limit cap (2^24 - 1)
+declare -r -i eip7825_gas_limit=16777215
+
+# Apply gas multiplier and check EIP-7825 limit
+# Usage: gas_limit="$(apply_gas_multiplier <gas_estimate>)"
+function apply_gas_multiplier {
+    declare -i _gas_estimate="$1"
+    shift
+
+    # Mantle has some real funky gas rules, exclude it from this logic
+    if (( chainid != 5000 )) && (( _gas_estimate > eip7825_gas_limit )) ; then
```

Comment:
We may need to exclude EraVm chains from this logic as well due to the 5:1 ergs:gas ratio. I haven't encountered this _yet_ but it's possible for the future.